algorithmic_efficiency/logging_utils.py

```diff
@@ -0,0 +1,492 @@
+from datetime import datetime
```

Comment:
nit: can you add a brief (even if redundant/obvious) file docstring?

Reply:
Done.